### PR TITLE
fix(training): honor TrainSpec.learning_rate in the lerobot_local trainer

### LINKED ISSUE #1020: TrainSpec.learning_rate is silently ignored by the lerobot_local trainer (every other backend honors it)
## Summary

`TrainSpec` exposes `learning_rate`, the `train_policy` tool forwards it as a first-class parameter, and four of the five training backends honor it — but the `lerobot_local` trainer never wires it into the config it builds, so the policy's training-preset LR is always used. A user who sets `learning_rate` is silently misled.

## Steps to reproduce

    from strands_robots.training import TrainSpec, create_trainer
    spec = TrainSpec(dataset_root=..., output_dir=..., learning_rate=5e-5,
                     extra={"policy_type": "act"})
    cfg = create_trainer("lerobot_local").build_config(spec)
    print(cfg.policy.optimizer_lr)   # -> 1e-05 (the ACT preset), not 5e-05

Run today on main: requested `5e-05` → `cfg.policy.optimizer_lr = 1e-05`, and `build_command()` emits no `--policy.optimizer_lr` flag either.

## Actual behavior

Confirmed on a real GPU run (two-phase ACT training, base + post-tune, Colab A100): both phases logged `lr:1.0e-05` (the ACT preset) regardless of the `TrainSpec.learning_rate` value.

## Root cause / asymmetry

- `training/lerobot.py` contains no reference to `learning_rate` — neither `build_config()` (typed path) nor `build_command()` (argv-parity path) maps it.
- Every other backend uses the field: PPO (`rl/ppo.py`) and FastSAC (`rl/fast_sac.py`) bind it to their Adam optimizers; GR00T forwards `--learning_rate`; Cosmos forwards `optimizer.lr`.
- The agent-facing `train_policy` tool exposes `learning_rate` as a parameter and passes it into the spec — so the agent path advertises a knob the lerobot backend ignores.
- A working escape hatch exists and is even exercised by the test suite (`extra={"policy.optimizer_lr": ...}`), which makes the ignored first-class field the odd one out.
- In fact `training/lerobot.py` reads **every other applicable `TrainSpec` field** (dataset_root / dataset_repo_id / streaming / base_model / output_dir / steps / global_batch_size / save_freq / num_gpus / num_nodes / resume / seed / method / lora_* / tune / val_episodes) — `learning_rate` is the only universal field with zero references in the file.

## Why this isn't the documented tolerance rule

`TrainSpec`'s docstring does let backends "read the fields they support and ignore the rest" — but fields with designed partial support say so on the field itself (`seed`: "best-effort; not all backends expose it"; `streaming`: "LeRobot-only"; `tune`: "Ignored by backends that don't" expose the toggles). `learning_rate` carries no such caveat, sits in the universal block, and is advertised by the `train_policy` tool — this reads as an accidental gap, not designed partial support. (A field-by-backend sweep found sibling gaps only in the Cosmos3 trainer — filing those separately.)

## Expected behavior

An explicit `learning_rate` reaches lerobot's optimizer config (`policy.optimizer_lr`, which the policy's `get_optimizer_preset()` reads) — or is rejected loudly. Silently ignoring it is the one behavior that can't be right; same class as #897 / #968 / #979 (make silent failure loud).

One design note: the field currently defaults to `1e-4`, so wiring it unconditionally would change every *default* lerobot run (the ACT preset is 1e-5). The natural shape is the `seed` pattern the spec already uses: default `None` = "the backend's own default", an explicit value = honored everywhere.

## Environment

main @eb48cdd · lerobot 0.5.1 · confirmed on Colab A100 (real run) and locally (config-level, Python 3.12)

### PR DESCRIPTION
**Pull request by cagataycali**
Fixes #1020

## Problem

`TrainSpec.learning_rate` was the only universal spec field with **zero references** in `training/lerobot.py`. `build_config()` never wired it into the typed `TrainPipelineConfig`, and `build_command()` emitted no `--policy.optimizer_lr` flag, so the policy's training preset always won and a caller-set value was silently dropped:

```python
spec = TrainSpec(dataset_root=..., output_dir=..., learning_rate=5e-5, extra={"policy_type": "act"})
cfg = create_trainer("lerobot_local").build_config(spec)
cfg.policy.optimizer_lr   # -> 1e-05 (the ACT preset), not the requested 5e-05
```

Every other backend honors the field (PPO/FastSAC bind it to their Adam optimizers, GR00T forwards `--learning_rate`, Cosmos forwards `optimizer.lr`), and the `train_policy` agent tool advertises it as a first-class parameter, so the lerobot path silently misled callers. Same class as the other make-silent-failure-loud fixes.

## Change

Make `learning_rate` opt-in, mirroring the `seed` pattern the spec already uses:

- **Default `None` = use the backend's own default** (the policy training preset for LeRobot, `FinetuneConfig` default for GR00T, TOML default for Cosmos). A plain run is unchanged.
- **An explicit value is honored by every backend.** For LeRobot it maps to `policy.optimizer_lr` (which the policy's `get_optimizer_preset()` reads) in both `build_config()` and `build_command()`. A policy config with no `optimizer_lr` field (its optimizer preset is not an Adam-style single LR) raises loudly instead of dropping the value.
- **RL trainers** (PPO/FastSAC) have no policy preset to defer to, so `RLTrainSpec` pins a concrete `1e-4` default in place of the base `None` — from-scratch RL behavior is preserved exactly.
- `train_policy` tool's `learning_rate` parameter defaults to `None` to match the honest semantics.
- GR00T / Cosmos now emit their LR override only when a value is set, so `None` cleanly falls through to their native defaults.

## Tests

New `TestLearningRate` in `tests/training/test_lerobot.py` (all fail on pre-fix `main`, pass after):

- explicit `learning_rate` reaches `cfg.policy.optimizer_lr` (pre-fix: `1e-05 == 5e-05` AssertionError)
- explicit LR is emitted as `--policy.optimizer_lr=` in `build_command`
- default `None` leaves the policy preset untouched (regression guard against changing every default run)
- default `None` emits no optimizer_lr flag
- explicit LR on a policy with no `optimizer_lr` field raises `ValueError`

RL and GR00T/Cosmos suites unchanged and green; the field is verified at the config level (as the report reproduces it). Docstrings on `TrainSpec.learning_rate` and the `train_policy` tool updated; CHANGELOG entry added.

### CODE REVIEW

**Review by yinsong1986 — APPROVED** [review]
## Summary

Wires `TrainSpec.learning_rate` into the `lerobot_local` trainer, which was the one universal spec field with zero references in `training/lerobot.py`. The fix makes the field opt-in (default `None` = defer to the backend's own default), mirroring the existing `seed` pattern: an explicit value now maps to `policy.optimizer_lr` in both `build_config()` and `build_command()`, and a policy config without an `optimizer_lr` field raises `ValueError` loudly rather than silently dropping the value. The base `TrainSpec.learning_rate` default flips `1e-4` -> `None`; `RLTrainSpec` re-pins a concrete `1e-4` (from-scratch RL has no preset to defer to), and GR00T/Cosmos now emit their LR override only when a value is set.

## What's good

- Behavior change is not silent: every consumer of `spec.learning_rate` (lerobot, groot, cosmos3, rl/ppo, rl/fast_sac) was checked and handles `None` correctly. RL paths take `RLTrainSpec` (pinned `1e-4`, never `None`); GR00T reads `learning_rate` from `FinetuneConfig` after the guarded kwargs build, so its default backstops the `None` case. No `None` ever reaches an optimizer constructor and no crash path was introduced.
- Follows AGENTS.md conventions: raise-on-fatal / no-silent-defaults (the `optimizer_lr` `hasattr` guard raises `ValueError`), opt-in shape consistent with `seed`, CHANGELOG entry added, dataclass field ordering valid (all fields defaulted).
- Scope is tight and the fix matches the described class of make-silent-failure-loud changes.
